nemo/utils/callbacks/nemo_model_checkpoint.py

```diff
@@ -198,6 +201,10 @@ def setup(self, trainer, pl_module, stage: str) -> None:
         self.last_model_path = trainer.strategy.broadcast(self.last_model_path)
 
     def on_save_checkpoint(self, trainer, pl_module, checkpoint):
+        # Load the correct state_dict for current checkpoint.
+        # Temporary solution.
+        if self.save_last_n_optim_states >= 0:
+            self._load_current_state_dict(trainer, checkpoint)
```

Comment:
why is this in `on_save_checkpoint`?

Reply:
Even if for some reason we need to put it here, let's make sure the `._load_current_state_dict` is not executed for "regular" saves